### ISSUE TITLE
Besides Blocks Lab 1 Feedback

### ISSUE THREAD

**Issue by lmock** [cs10/bjc-r#306]
Hey everyone - please leave your feedback for the first lab of BB here. Thank you! :)

**Comment by lmock** [comment]
Sumer: On the second to last page of the lab, it doesn't have anything for "show answer", though I'm not sure if that is just because of skulpt.

**Comment by lmock** [comment]
@peterasujan : Looks like it's not running some javascript... If you want to see the solution, it should show up here: http://beautyjoy.github.io/bjc-r/topic/topic.html?topic=berkeley_bjc/python/besides-blocks-intro.topic&noassignment=true&noreading=true&novideo=true&course=cs10_sp14.html.

**Comment by lmock** [comment]
@ddgarcia : I see the recursion page, with no Python-in-the-browser place to
write code.

Only "click here to show answer" which does nothing.

Also, many places where python code is not formatted in courier.

Also "Besides Python: Data Redux" should be called "Besides Python part I"
or something

Also "Courseware" "Course Info" "Discussion" "Progress" "Weekly Schedule and Office Hours" "Instructor" seem more confusing than needed. Can they be removed?

Also I'm in Recursion and can't seem to move out of it to go back.

Also when we're showing interaction with the interpreter,
code that's typed in should be bold and comments in italics
to make it easier to read. See
http://www-inst.eecs.berkeley.edu/~selfpace/cs9honline/#QuizExpressions
for an example of how we show it in the 9H class.

**Comment by peterasujan** [comment]
>  Looks like it's not running some javascript

Indeed, looking at the page source shows that the javascript to toggle the visibility is gone. Could this be being accidentally stripped in the build process?

**Comment by songsok** [comment]
Somehow edX ignores the JavaScript. It comments out the JavaScript

**Comment by jasnowiss** [comment]
Nit-picky, but missing a close parentheses in the hi(x) definition on page 5.
Also, when I clicked the 'click to show answer' button in the recursion page, all the links to the other pages broke. It won't let me go anywhere. Hopefully you guys are just doing maintenance or something.

**Comment by songsok** [comment]
The button doesn't work because edX somehow commented out the JavaScript

**Comment by peterasujan** [comment]
Hmm that's pretty weird..
One temporary fix would be to make it so that the solutions are just always displayed - i.e. remove the <code>style = "display: none"</code> from the div containing your answer.

**Comment by marwahaha** [comment]
Hi all,
The freezing issue is fixed. hi(x) closing paren is fixed.

**Comment by songsok** [comment]
So, on the page "Loops", when I wrote the min(x, y, z) function and run it, it doesn't seem to give back a result. I ran that function in python on the terminal and it seems to work fine. So it's not the function that doesn't work it seems more like Skulpt still has some issue. 

edit: so I tried to print out the function (print(min(1, 2, 3)) and it actually works. So it seems like we have to print everytime we want to see values from function call.

**Comment by lmock** [comment]
Typo on string manipulation page -- In Python, there is something called "string". It's not the kind of string that you might think of. A string in Python is a way of representing text. A string is denoted by "" or ''. 

The last part of the sentence.

**Comment by marwahaha** [comment]
Added a space between each set of quotes.

**Comment by cycomachead** [comment]
Caren's changes need to be integrated:
https://github.com/beautyjoy/bjc-r/pull/301/files

Also, I'm not sure if all comments from this were fixed:
https://github.com/beautyjoy/bjc-r/pull/302/

**Comment by peterasujan** [comment]
Some python-related issues I've been running into:
- Skulpt is super annoying and picky - it gives no error output, which makes it nearly impossible to detect things like indentation errors. Pressing tab moves to the next field, rather than indenting, which makes me sad.
- Speaking of indentation, the current python 3 setup on the lab machines doesn't allow indentation by pressing tab - it passes the tab to the shell itself and tries to do tab-completion

**Comment by cycomachead** [comment]
What, even in terminal? This is a bug. It doesn't do that over ssh.

I'm trying to fix skulpt...

## 

Michael Ball
http://www.michaelballphoto.com
Photos & iPhone Apps

**Comment by cycomachead** [comment]
Wait, I take that back. Something changed in the past couple days.

Dammit, it's a python 3.4 bug with bash and OS X. 3.3.3 and 2.7.5 don't do
it.

## 

Michael Ball
http://www.michaelballphoto.com
Photos & iPhone Apps

**Comment by jasnowiss** [comment]
Typo in the loops page.

> > > add_one_n_times(-5):
> > > -5 
> > > add_one_n_times(5):
> > > 5 

The colons at the end of the function calls should not be there.

**Comment by songsok** [comment]
Yeah I saw that.. I'm working on it

**Comment by jasnowiss** [comment]
On the strings manipulation page, change the parameters of the extra_end and cat_dog functions to be something besides 'str'. Python treats 'str' differently, so it's acting strangely when students start writing the function with it.
Also, for the built-in functions on the same page, replace the "e" in the arr.index example with something else, maybe "r"? It's a little confusing that this example and the example before it both return 1. Students might misunderstand what index does.

**Comment by jasnowiss** [comment]
Nevermind on the parameters thing, it still works even if str is the parameter.

**Comment by cycomachead** [comment]
> Nevermind on the parameters thing, it still works even if str is the parameter.

It should work, but we SHOULD change the example! It's terrible practice, especially if you try calling `str()` in that function.

**Comment by songsok** [comment]
I made all the changes already, if you still run into problems, give more
feedback

**Comment by cycomachead** [comment]
On edx?

We should be making changes in the repo.

**Comment by songsok** [comment]
Ok will do that too

**Comment by itssumertime** [comment]
The lab is missing an explanation for "not equals" being "!=" so if that could be in the chart it would clear up a lot of confusion.

**Comment by songsok** [comment]
Ok will do

**Comment by ibirnam** [comment]
1) skulpt did not serve as an effective replacement for an interpreter. You couldn't use tab within it, and students got confused that they had to print stuff to see anything. For example if students type the following into skuplt:
def sum(x,y):
    return x + y
sum(3,4)
and then hit the Run button, they dont see anything, where as if they did this in the terminal, the function would be evaluated, so it was a bit confusing since skulpt didn't evaluate AND show the result simultaneously, requiring a print call to show results.

2) import turtle doesn't work in python3 in the terminal on the lab machines.

3) Was said before, but no explanation on what != means

4) Great job otherwise!!

**Comment by lmock** [comment]
Loops page:
The range function can also be written with one argument like this: range(y):,
This just means that you're not explicitly telling it to start from a specific number but rather from 0 to y - 1

No punctuation after y - 1

**Comment by lmock** [comment]
Loops page:

There are two essential kinds of loop in (also in most of other text programming languages as well) Python, they are For-Loop and While-Loop.

Grammar errors

**Comment by lmock** [comment]
Also these hash marks need to GO AWAY. They look really ugly in edX.

I know they're delineating comments, but there must be another way to show comments than to use hashes.

**Comment by lmock** [comment]
There's no explanation of what -= means (you see it first on the Loops page). If you're going to mention += in the beginning, you should just stick it somewhere obvious / explain it to people.

**Comment by lmock** [comment]
There needs to be a better distinct visual explanation division of the for-loop and while-loop. The content on the page looks mashed together. For example, the Functions in Python page has a clear way of dividing content - head and body.
